crates/quote/src/js_api/impls.rs

```diff
@@ -106,11 +107,67 @@ impl From<crate::QuoteResult> for super::QuoteResult {
     }
 }
 
+impl From<MainPair> for Pair {
+    fn from(value: MainPair) -> Self {
+        Pair {
+            pair_name: value.pair_name,
+            input_index: value.input_index,
+            output_index: value.output_index,
+        }
+    }
+}
+impl From<Pair> for MainPair {
+    fn from(value: Pair) -> Self {
+        MainPair {
+            pair_name: value.pair_name,
+            input_index: value.input_index,
+            output_index: value.output_index,
+        }
+    }
+}
+
+impl From<MainBatchOrderQuotesResponse> for BatchOrderQuotesResponse {
+    fn from(value: MainBatchOrderQuotesResponse) -> Self {
+        let mut block_number_error = "block number, ".to_string();
```

Comment:
what is this? why can't we use thiserror and use formatting like we do with all our other errors?